forms-flow-api/src/formsflow_api/services/import_support.py

```diff
@@ -0,0 +1,343 @@
+        return mapper
+
+    @user_context
+    def import_form_workflow(self, request, **kwargs):  # pylint: disable=too-many-locals, too-many-statements, too-many-branches
+        """Import form/workflow."""
+        current_app.logger.info("Processing import data...")
+        request_data, file = self.validate_input_data(request)
+        schema = ImportRequestSchema()
+        input_data = schema.load(request_data)
+        import_type = input_data.get("import_type")
+        action = input_data.get("action")
+        user: UserContext = kwargs["user"]
+        tenant_key = user.tenant_key
+
+        # Check if the action is valid
+        if action not in ["validate", "import"]:
+            raise BusinessException(BusinessErrorCode.INVALID_INPUT)
+
+        if import_type == "new":
+            current_app.logger.info("Import new processing..")
+            # Validate input file type whether it is json
+            if not self.validate_file_type(file.filename, (".json",)):
+                raise BusinessException(BusinessErrorCode.INVALID_FILE_TYPE)
+            current_app.logger.info("Valid json file type.")
+            file_data = self.read_json_data(file)
+            # Vaidate file data
+            self.validate_input_json(file_data, form_workflow_schema)
+            form_json = file_data.get("forms")[0].get("content")
+            workflow_data = file_data.get("workflows")[0].get("content")
+            # Check the form name/path/title already exists
+            query_params = (
+                f"title={form_json.get('title')}&"
+                f"name={form_json.get('name')}&"
+                f"path={form_json.get('path')}&"
+                "select=title,path,name"
+            )
+            self.valide_form_exists(query_params)
+            if action == "validate":
+                # On Import new, version will be 1.0
+                return self.version_response(
+                    form_major=1, form_minor=0, workflow_major=1, workflow_minor=0
+                )
+            if action == "import":
+                self.import_new_form_workflow(file_data, form_json, workflow_data)
+        else:
+            current_app.logger.info("Import edit processing..")
+            edit_request = ImportEditRequestSchema().load(request_data)
+            valid_file = self.validate_file_type(file.filename, (".json", ".bpmn"))
+            mapper_id = edit_request.get("mapper_id")
+
+            if valid_file == ".json":
+                file_data = self.read_json_data(file)
+                # Validate input json file whether only form or form+workflow
+                try:
+                    validate(instance=file_data, schema=form_schema)
+                    current_app.logger.info("Form only import inprogress...")
+                    # TODO Only form import # pylint: disable=fixme
+                except ValidationError:
+                    self.validate_input_json(file_data, form_workflow_schema)

```

Comment:
Can we move logic from except block to keep it clean ?

Reply:
done